### PR TITLE
Move declarations test to GA

### PR DESCRIPTION
**Pull request by alexserov**

### CODE REVIEW

**Review by nightskylark — DISMISSED** [review]

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -66,6 +66,26 @@ jobs:
     - name: Lint .d.ts
       run: npm run lint-dts
 
+    - name: Validate declarations
+      run: |
+        target=./ts/dx.all.d.ts
+        cp $target $target.current
+
+        npm ls devextreme-internal-tools || :
```
Comment (pavelgruba) [inline .github/workflows/lint.yml]:
```suggestion
```

File: docker-ci.sh [hunk 2]
```diff
@@ -13,25 +13,9 @@ function run_lint {
...
-        cat $target.diff | sed "1,2d"
-        exit 1
-    else
-        echo "TS is up-to-date"
-    fi
-
-    npm run validate-ts
+    echo 'Skipped'
+    # This function is a stub for a Drone; it is a subject to remove in future
+    # Current implementation located in .github/workflows/lint.yml
```
Comment (pavelgruba) [inline docker-ci.sh]:
```suggestion
    # Current implementation located in .github/workflows/ts_declarations.yml
```

File: .github/workflows/ts_declarations.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        
+        if ! diff $target.current $target -U 5 > $target.diff; then
+            echo "FAIL: $target is outdated:"
+            cat $target.diff | sed "1,2d"
+            exit 1
+        else
+            echo "TS is up-to-date"
+        fi
+        
+        npm run validate-ts
```
Comment (IlyaKhD) [inline .github/workflows/ts_declarations.yml]: I suggest splitting this into 2 jobs:

#### check-ts-bundle:
```yaml
        npm run update-ts
        
        if ! diff $target.current $target -U 5 > $target.diff; then
            echo "FAIL: $target is outdated:"
            cat $target.diff | sed "1,2d"
            exit 1
        else
            echo "TS is up-to-date"
        fi
```


#### validate-declarations:

```yaml
npm run validate-ts
npm run validate-declarations
```

File: .github/workflows/ts_declarations.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      uses: actions/cache@v2
+      with:
+        path: ./node_modules
+        key: ${{ runner.os }}-node-modules-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: ${{ runner.os }}-node-modules
+
+    - name: Run npm install
+      run: npm install --no-audit --no-fund
+
+    - name: Validate declarations
```
Comment (IlyaKhD) [inline .github/workflows/ts_declarations.yml]:
```suggestion
    - name: Check dx.all.d.ts is up-to-date
```

**Review by IlyaKhD — APPROVED** [review]

**Review by pavelgruba — APPROVED** [review]

**Pull request merged**